torch/_dynamo/compiled_autograd.py

```diff
@@ -147,6 +151,71 @@ def begin_capture(
         self.stack.enter_context(preserve_node_meta())
         return inputs, sizes, scalars
 
+    def proxy_call_lambda(
+        self,
+        idx,
+        inputs,
+        output_metadatas,
+    ):
+        with disable_proxy_modes_tracing():
+            # create fake Tensors
+            grad_ins: List[Optional[torch.Tensor]] = []
+            for output_metadata in output_metadatas:
+                if output_metadata is None:
+                    grad_ins.append(None)
+                    continue
+
+                layout, device, dtype, size = output_metadata
+                grad_ins.append(
+                    torch.empty(size=size, dtype=dtype, layout=layout, device=device)
+                )
+
+            global next_op
+            @torch.library.custom_op(f"compiled_autograd::cpp_node_op_{next_op}", mutates_args=())
+            def cpp_node_op_i(inputs: List[torch.Tensor], idx: int) -> List[torch.Tensor]:
+                breakpoint()
+                outs = torch._C._dynamo.compiled_autograd.call_lambda(inputs, idx)
+                return [out.clone() for out in outs]
+            
+            op_outs = [x.clone().detach() for x in grad_ins]
+
+            @cpp_node_op_i.register_fake
+            def _(inputs, idx):
+                grad_ins: List[Optional[torch.Tensor]] = []
+                for output_metadata in output_metadatas:
+                    if output_metadata is None:
+                        grad_ins.append(None)
+                        continue
+
+                    layout, device, dtype, size = output_metadata
+                    grad_ins.append(
+                        torch.empty(size=size, dtype=dtype, layout=layout, device=device)
+                    )
```

Comment:
This isn't correct w.r.t. strides and Tensor Subclasses. I'm not sure if we can work around those.

- This returns grad_ins with contiguous strides. This is not always the case, these nodes can return non-contiguous strides
- If any of the inputs are Tensor Subclasses, then the grad_ins might also need to be Tensor Subclasses. The logic here doesn't allow for that to happen.

Reply:
Will these affect which ops get called downstream in the graph?

Should be fixable as long as they are info we can infer from the forward pass's inputs

Reply:
> Is there also a contract that they must match the forward pass's inputs?

No

Reply:
From offline discussion: we could assume the strides and then later on (at runtime) assert them or coerce them, similar to how AOTAutograd does it. Could also theoretically do this for subclassness.